File: 백준/Silver/25631. 마트료시카 합치기/마트료시카 합치기.java
import java.io.*;
import java.sql.Array;
import java.util.*;



public class Main {
    public static void main(String[] args) throws NumberFormatException, IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());

        ArrayList<Integer> arr = new ArrayList<>();

        while(st.hasMoreTokens()){
            arr.add(Integer.parseInt(st.nextToken()));
        }

        Collections.sort(arr);


        int ans = 0;
        while(true){

            int temp = 0;
            int cnt = 0 ;
            ans ++ ;

            for(int i = 0 ; i < arr.size(); i++){
                if(temp < arr.get(i)){
                    temp = arr.get(i);
                    //System.out.println(arr.get(i));
                    arr.set(i,0);
                }
                if(arr.get(i)==0){
                    cnt++;
                }
                
            }

            if( cnt == n){
                break;
            }


        }
        System.out.println(ans);







    }
}

File: 백준/Silver/25631. 마트료시카 합치기/마트료시카 합치기.cc
#include <bits/stdc++.h>
using namespace std;

int main() {

    ios_base::sync_with_stdio(false);
    cout.tie(0);
    cin.tie(0);

    int n;
    cin >> n;

    vector<int>v;

    for(int i = 0 ; i < n ; i++){
        int a;
        cin >> a;
        v.push_back(a);
    }

    sort(v.begin(),v.end());

    int ans = 0 ;
    while(true){

        int temp =0;
        int cnt = 0;
        ans ++;
        for(int i = 0 ; i < v.size(); i++){

            if(temp < v[i]){
                temp = v[i];
                v[i]=0;
            }

            if(v[i]==0){
                cnt++;
            }
        }

        if(cnt == v.size()){
            break;
        }

    }
    cout << ans << "\n";








}

